min-turnup/phase1/azure/lib/azure.jsonnet

```diff
@@ -0,0 +1,285 @@
+function(cfg1)
+	local cfg = cfg1.phase1;
+	local master_private_ip = cfg.azure.master_private_ip;
+	local names = {
+		resource_group: "%(instance_prefix)s" % cfg,
+		master_public_ip: "%(instance_prefix)s-master-pip" % cfg,
+		availability_set: "%(instance_prefix)s-as" % cfg,
+		storage_account: "${replace(\"%(instance_prefix)sstrg\", \"-\", \"\")}" % cfg,
+		storage_container: "kube0000%(instance_prefix)s" % cfg,
+		vnet: "%(instance_prefix)s-vnet" % cfg,
+		subnet: "%(instance_prefix)s-subnet" % cfg,
+		route_table: "%(instance_prefix)s" % cfg,
+		security_group: "%(instance_prefix)s-nsg" % cfg,
+		master_nic: "%(instance_prefix)s-master-nic" % cfg,
+		master_vm: "%(instance_prefix)s-master" % cfg,
+		node_nic: "%(instance_prefix)s-node-nic" % cfg,
+		node_vm: "%(instance_prefix)s-node" % cfg,
+	};
+	{
+		variable: {
+			subscription_id: { default: cfg.azure.subscription_id },
+			tenant_id: { default: cfg.azure.tenant_id },
+			client_id: { default: cfg.azure.client_id },
+			client_secret: { default: cfg.azure.client_secret },
+		},
+		provider: {
+			azurerm: {
+				subscription_id: "${var.subscription_id}",
+				tenant_id: "${var.tenant_id}",
+				client_id: "${var.client_id}",
+				client_secret: "${var.client_secret}",
+			}
+		},
+		resource: {
+			azurerm_resource_group: {
+				rg: {
```

Comment:
So imagine a jsonnet input for the `azurerm_resource_group` resource would look like:

```
{
  azurerm_resource_group: resource_map([
    {
      name: cfg.instance_prefix,
    },
  ])
}
```

Later on, since all resource wind up "parented" to a resource group, I have to do:

```
{
  azurerm_blahblah_resource: resource_map([
    {
      resource_group_name: "${azurerm_resource_group." + cfg.instance_prefix + ".name}"
    },
  ])
}
```

I don't want to cheat and just use `"resource_group_name": cfg.instance_prefix` because it creates a non-explicit dependency; aka things would break if someone ever modified the Terraform template to append `-rg` in the `azurerm_resource_group.?.name` field. That's sort of why I chose the shorthand names in the first place.

That having been said, that syntax isn't really a big deal. The suggested change is probably more discoverable for someone looking at their terraform output, with what I have now they might think "`azurerm_resource_group.rg`... what is `rg`? My resource group is named `cfg.instance_prefix`".

After typing all of this, I've sort of convinced myself to go with the suggestion. Let me know if you guys have other thoughts.